packs/hpe-icsp/actions/lib/icsp.py

```diff
@@ -63,14 +63,31 @@ def get_sessionid(self):
 
         # added here due to the requirement of the session id
         self.base_headers = {'Auth': self.icsp_sessionid,
-                   'X-Api-Version': self.icsp_apiv}
+                             'X-Api-Version': self.icsp_apiv}
 
         return results["sessionID"]
 
     def extract_id(self, joburi):
         jobid = str(joburi)
         return int(jobid.split("/")[-1])
 
+    def get_mids(self, ids, idtype):
+        endpoint = "/rest/os-deployment-servers"
+        getresults = self.icsp_get(endpoint)
+        servers = getresults["members"]
+        mids = []
+        for id in ids:
+            for server in servers:
+                if idtype == 'serialnumber':
+                    if ((server["serialNumber"] == id) and
```

Comment:
Ah, ok, thanks for clarifying it.

Probably also a good idea to add the same comment to the code - aka why it's done the way it is :)